modules/distribution/resources/operation_policies/specifications/opaPolicy.json

```
@@ -0,0 +1,94 @@
+{
+  "category": "Mediation",
+  "name": "opaPolicy",
+  "displayName": "Validate Request With OPA Policy",
+  "description": "With this policy, user can validate requests based on the OPA policy engine",
+  "multipleAllowed": true,
+  "policyAttributes": [
+    {
+      "name": "serverUrl",
+      "displayName": "OPA server URL",
+      "description": "OPA server's URL",
+      "validationRegex": "([\\w+]+\\:\/\/)?([\\w\\d-]+\\.)*[\\w-]+[\\.\\:]\\w+([\/\\?\\=\\&\\#\\.]?[\\w-]+)*\/?",
+      "type": "String",
+      "required": true
+    },
+    {
+      "name": "accessKey",
+      "displayName": "Access Key",
+      "description": "Access key for validation request",
+      "validationRegex": "([a-zA-Z\\d_][a-zA-Z\\d_\\-\\ ]*)$",
+      "type": "String",
+      "required": false
+    },
+    {
+      "name": "policy",
+      "displayName": "Policy",
+      "description": "Policy to be validated",
+      "validationRegex": "([a-zA-Z\\d_][a-zA-Z\\d_\\-\\ ]*)$",
+      "type": "String",
+      "required": true
+    },
+    {
+      "name": "rule",
+      "displayName": "Rule",
+      "description": "Rule to validate",
+      "validationRegex": "([a-zA-Z\\d_][a-zA-Z\\d_\\-\\ ]*)$",
+      "type": "String",
+      "required": false
+    },
+    {
+      "name": "sendAccessToken",
+      "displayName": "Send access token",
+      "description": "Select whether to send access token to the OPA payload",
+      "validationRegex": "^(true|false)$",
+      "type": "Boolean",
+      "defaultValue": false,
+      "required": false
+    },
+    {
+      "name": "additionalMCProperties",
+      "displayName": "Additional MC properties",
+      "description": "Additional message context properties to be included in the OPA input. Add these properties in a comma seperated list.",
```

Shall we have a common name for CC and synapse. wdyt?

```suggestion
      "name": "additionalProperties",
      "displayName": "Additional properties",
      "description": "Additional message context (request context) properties to be included in the OPA input. Add these properties in a comma seperated list.",
```